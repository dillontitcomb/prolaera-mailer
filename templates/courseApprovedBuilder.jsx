import React from 'react';
import { Email, renderEmail } from 'react-html-email';
import CourseApproved from '../components/courseApproved/courseApproved';
import CourseApprovedSubHeader from '../components/courseApproved/courseApprovedSubHeader';
import Footer from '../components/footer';
import builderHeader from '../components/header';
import css from './templateCSS.js';

const courseApprovedBuilder = async imageUrl => {
  try {
    const Header = await builderHeader(imageUrl);
    return renderEmail(
      <Email title="Course approved on Prolaera" headCSS={css}>
        <Header />
        <CourseApprovedSubHeader />
        <CourseApproved />
        <Footer />
      </Email>
    );
  } catch (error) {
    throw error;
  }
};

export default courseApprovedBuilder;
