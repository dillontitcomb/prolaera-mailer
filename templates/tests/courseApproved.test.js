import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import fs from 'fs';
import React from 'react';
import renderer from 'react-test-renderer';
import CourseApproved from '../../components/courseApproved/courseApproved';
import courseApprovedBuilder from '../courseApprovedBuilder';

Enzyme.configure({ adapter: new Adapter() });

const user = {
  email: 'test@email.com'
};
const course = {
  by: 'John Doe',
  course_id: '1234-1234-1234-1234',
  courseAudience: 'Accountants',
  delivery_method: 1,
  hours: [
    {
      credits: 1,
      subject_area: 'Tax Preparation'
    }
  ],
  level: 'Basic',
  name: 'Tax Preparation 102',
  objectives: '<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>',
  prep: '<p>Read course introduction</p>',
  prerequisites: '<p>Tax Prep 101</p>',
  summary: '<p>This course is the second installation of Tax Prep focused on general tax planning.</p>'
};

describe('buildHeader image resizing tests', () => {
  it('writes CourseApproved html file to courseApprovedTest.html', async () => {
    const courseApprovedEmail = await courseApprovedBuilder(
      'https://assets.prolaera.com/prolaeraLogo_fullText.png',
      user,
      course
    );
    const saved = await writeFile(courseApprovedEmail);
    expect(saved).toEqual(true);
  });
});

describe('CourseApproved component', () => {
  it('creates and checks snapshot of CourseApproved component html', async () => {
    const approvedHtml = renderer.create(<CourseApproved {...user} {...course} />);
    const approvedJson = approvedHtml.toJSON();
    expect(approvedJson).toMatchSnapshot();
  });
  it('checks default CourseApproved html', async () => {
    const wrapper = shallow(<CourseApproved {...user} {...course} />);
    expect(
      wrapper.contains(
        <p>
          Tax Preparation 102 has been reviewed and approved by an admin at your firm and is now in the marketplace!
        </p>
      )
    ).toBe(true);
  });
  it('checks custom course name in CourseApproved html', async () => {
    const wrapper = shallow(<CourseApproved {...user} {...course} name={'Test Course'} />);
    expect(
      wrapper.contains(
        <p>Test Course has been reviewed and approved by an admin at your firm and is now in the marketplace!</p>
      )
    ).toBe(true);
  });
  it('checks custom button link in CourseApproved html', async () => {
    const wrapper = shallow(<CourseApproved {...user} {...course} buttonLink={'https://www.testLink.com'} />);
    expect(
      wrapper.contains(
        <a
          href="https://www.testLink.com/test@email.com"
          className="accountButton"
          style={{
            maxWidth: '200px',
            backgroundColor: '#68B63E',
            padding: '14px 14px 14px 14px',
            color: 'white',
            textDecoration: 'none'
          }}
        >
          Log In
        </a>
      )
    ).toBe(true);
  });

  it('checks custom user email in CourseApproved html', async () => {
    const wrapper = shallow(<CourseApproved {...user} {...course} email={'test@email.com'} />);
    expect(
      wrapper.contains(
        <a
          href="www.exampleButtonLink.com/test@email.com"
          className="accountButton"
          style={{
            maxWidth: '200px',
            backgroundColor: '#68B63E',
            padding: '14px 14px 14px 14px',
            color: 'white',
            textDecoration: 'none'
          }}
        >
          Log In
        </a>
      )
    ).toBe(true);
  });
});

async function writeFile(emailHtml) {
  return new Promise((resolve, reject) => {
    fs.writeFile(`${__dirname}/courseApprovedTest.html`, emailHtml, err => {
      if (err) {
        console.log(err);
        reject(err);
      }
      resolve(true);
    });
  });
}
