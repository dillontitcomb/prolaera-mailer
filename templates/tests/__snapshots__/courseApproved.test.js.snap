// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseApproved component creates and checks snapshot of CourseApproved component html 1`] = `
<div
  style={
    Object {
      "border": "1px solid lightgray",
      "display": "block",
      "fontFamily": "Arial, sans-serif",
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": "20px",
      "textAlign": "center",
      "width": "500px",
    }
  }
>
  <table
    align="center"
    border="0"
    cellPadding={0}
    cellSpacing={0}
    className="courseApproved"
    valign="top"
    width="500px"
  >
    <tbody>
      <tr>
        <td
          align="left"
        >
          <p>
            Tax Preparation 102
             has been reviewed and approved by an admin at your firm and is now in the marketplace!
          </p>
          <div
            className="buttonContainer"
            style={
              Object {
                "padding": "20px 0px 20px 0px",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <a
              className="accountButton"
              href="www.exampleButtonLink.com/test@email.com"
              style={
                Object {
                  "backgroundColor": "#68B63E",
                  "color": "white",
                  "maxWidth": "200px",
                  "padding": "14px 14px 14px 14px",
                  "textDecoration": "none",
                }
              }
            >
              Log In
            </a>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
